python/test/unit/language/test_core.py

```diff
@@ -7526,7 +7529,7 @@ def unsplat_kernel(x, explicit: tl.constexpr):
         if explicit:
             condition = condition.reshape([])
 
-        if condition:
+        if condition.reshape([]):
```

Comment:
I think `explicit=False` case is there to test the implicit support of ndim conditions. Maybe just add a warning filter to the test?